packages/vite/src/node/plugins/manifest.ts

```diff
@@ -128,17 +129,18 @@ export function manifestPlugin(config: ResolvedConfig): Plugin {
           manifest[getChunkName(chunk)] = createChunk(chunk)
         } else if (chunk.type === 'asset' && typeof chunk.name === 'string') {
           // Add every unique asset to the manifest, keyed by its original name
-          const src =
-            fileNameToAssetMeta.get(chunk.fileName)?.originalName ?? chunk.name
-          const asset = createAsset(chunk, src)
+          const assetMeta = fileNameToAssetMeta.get(chunk.fileName)
+          const src = assetMeta?.originalName ?? chunk.name
+          const asset = createAsset(chunk, src, assetMeta?.isEntry)
           manifest[src] = asset
           fileNameToAsset.set(chunk.fileName, asset)
         }
       }
 
-      // Add duplicate assets to the manifest
-      fileNameToAssetMeta.forEach(({ originalName }, fileName) => {
+      // Add deduplicated assets to the manifest
+      assets.forEach(({ originalName }, referenceId) => {
```

Comment:
I made a mistake in the prev PR here, `fileNameToAssetMeta` doesn't have all `assets`. It worked because the test had a single duplicate. I can move this part to a separate PR if you think it is more clear. I consider this PR and the prev part of the same refactoring though.

Reply:
I'm a bit confused why `fileNameToAssetMeta` doesn't have all of `assets`? Above you did:

```ts
const fileNameToAssetMeta = new Map<string, GeneratedAssetMeta>()
      const assets = generatedAssets.get(config)!
      assets.forEach((asset, referenceId) => {
        const fileName = this.getFileName(referenceId)
        fileNameToAssetMeta.set(fileName, asset)
      })
```

so theoretically it should have all of `assets`?

Reply:
Because the deduplicated assets end up with the same `fileName` as others

Reply:
Ah right we want to duplicate the assets when generating the manifest. Thanks for the explanation!